blockchain/genesis/genesis.go

```diff
@@ -325,8 +327,20 @@ func Timestamp() int64 {
 	return atomic.LoadInt64(&genesisTs)
 }
 
-// Hash is the hash of genesis config
-func (g *Genesis) Hash() hash.Hash256 {
+// LoadGenesisHash is done once to compute and save the genesis block's hash
+func (g *Genesis) LoadGenesisHash() {
+	loadGenesisHash.Do(func() {
+		genesisHash = g.ComputeHash()
+	})
+}
+
+// Hash returns the genesis block's hash
+func Hash() hash.Hash256 {
+	return genesisHash
+}
+
+// ComputeHash computes the hash of genesis config
+func (g *Genesis) ComputeHash() hash.Hash256 {
```

Comment:
no conflict to call it "Hash"

Reply:
yes, no conflict, but i feel it is quite confusing? you can have a variable
```
var g genesis.Genesis
hash := g.Hash()  // this computes the hash
```
and a package level
```
hash := genesis.Hash() // this reads the global variable
```
so i renamed to ComputeHash